Pour develop into master

3m3kalionel opened the pull request.

3m3kalionel committed:
Bug fixing using instrumenter

3m3kalionel committed:
Edit readme file

3m3kalionel committed:
Ignore coverage directory

3m3kalionel committed:
Edit file name

3m3kalionel committed:
Edit test script command

3m3kalionel committed:
Edit config

3m3kalionel committed:
Fix merge conflict in .gitignore file

3m3kalionel committed:
Merge branch 'develop' into feature/tests

3m3kalionel committed:
Merge pull request #2 from 3m3kalionel/feature/tests

Feature/tests

3m3kalionel committed:
Merge pull request #3 from 3m3kalionel/feature/tests

Feature/tests

3m3kalionel committed:
Commit pull request changes

3m3kalionel committed:
Fix merge conflict

3m3kalionel committed:
Merge pull request #4 from 3m3kalionel/server-side

Server side

3m3kalionel committed:
Implement belongs-to-many association

3m3kalionel committed:
Implement many-to-many association

3m3kalionel committed:
Edit syntax

3m3kalionel committed:
Set unique email for user

3m3kalionel committed:
Remove unique property

3m3kalionel committed:
Group creation Bug fixes

3m3kalionel committed:
Refactor code using latest syntax

3m3kalionel committed:
Delete user-group migrations file

3m3kalionel committed:
Delete references field to fix userId and groupId bugs

3m3kalionel committed:
Edit start command

3m3kalionel committed:
Delete main field

3m3kalionel committed:
Fix start bug

3m3kalionel committed:
Synchronize tables not yet in the database

3m3kalionel committed:
Fix UserId and GroupId not found bugs

3m3kalionel committed:
Edit model associations using new syntax

3m3kalionel committed:
Edit model associations using new syntax

3m3kalionel committed:
Add constant values for addMember function

3m3kalionel committed:
Add route to allow users add group members

3m3kalionel committed:
fix user not found bug

3m3kalionel committed:
Remove debugging comments

3m3kalionel committed:
Merge pull request #5 from 3m3kalionel/server-side

Server side

Commit:
Return created group

Commit:
Fix create new group test bugs

Commit:
Fix create new group test bugs

Commit:
Fix post message route bugs

Commit:
Add auto-generated file

Commit:
Add a userId property

Commit:
Add test fixture

Commit:
Add test for message model

Commit:
Add tests for user model

Commit:
Set test timeout

Commit:
git reset HEAD

Commit:
Edit the catch block

Commit:
Delete response statement

3m3kalionel committed:
Edit model name to fix bug

3m3kalionel committed:
Remove catch block

3m3kalionel committed:
Add empty line at the end of file

3m3kalionel committed:
Fix user not found bug

3m3kalionel committed:
Remove log statements

3m3kalionel committed:
Edit port number

3m3kalionel committed:
Remove eslint error flags

3m3kalionel committed:
Remove eslint flag errors

3m3kalionel committed:
Merge pull request #6 from 3m3kalionel/feature/tests

Feature/tests

3m3kalionel committed:
Fix error in route for retrieving messages

3m3kalionel committed:
Edit status code error

3m3kalionel committed:
Merge pull request #7 from 3m3kalionel/feature/tests

Feature/tests

3m3kalionel committed:
Delete commented code

3m3kalionel committed:
Delete bug fixing statements

3m3kalionel committed:
Fix eslint flag errors

3m3kalionel committed:
Refactor code

3m3kalionel committed:
Edit response error message

3m3kalionel committed:
Catch User sync model error

3m3kalionel committed:
Add password validation

3m3kalionel committed:
Add email validation

3m3kalionel committed:
Add limit on entry passwords

3m3kalionel committed:
Edit error response mesage

3m3kalionel committed:
Refactor code

3m3kalionel committed:
set up test template

3m3kalionel committed:
Add validUser object

3m3kalionel committed:
Set test environment

3m3kalionel committed:
Edit error response

3m3kalionel committed:
Edit test script

3m3kalionel committed:
Edit eslint flag error

3m3kalionel committed:
Remove eslint error flag

3m3kalionel committed:
Add user sign up tests

3m3kalionel committed:
Add dummy test data

3m3kalionel committed:
Merge pull request #8 from 3m3kalionel/feature/tests

Feature/tests

3m3kalionel committed:
Edit database name

3m3kalionel committed:
Merge pull request #9 from 3m3kalionel/feature/tests

Edit database name

3m3kalionel committed:
Add a migrate-test command

3m3kalionel committed:
Add a before_script test command

3m3kalionel committed:
Merge pull request #10 from 3m3kalionel/feature/tests

Feature/tests

3m3kalionel committed:
Ignore .nyc_output

3m3kalionel committed:
Edit coverage task

3m3kalionel committed:
Add user login tests

3m3kalionel committed:
Run coverage after build

3m3kalionel committed:
Add custom script commands

3m3kalionel committed:
Remove redundant code

3m3kalionel committed:
Add signup tests

3m3kalionel committed:
Ignore .DS_Store file

3m3kalionel committed:
Merge pull request #11 from 3m3kalionel/feature/tests

Feature/tests

3m3kalionel committed:
Move babel-register from dev-dependencies to dependencies

3m3kalionel committed:
Move babel-preset-es-2015 to dependencies

3m3kalionel committed:
Setup production database

3m3kalionel committed:
Edit root route response

3m3kalionel committed:
Add badges to readme

3m3kalionel committed:
Merge pull request #12 from 3m3kalionel/feature/tests

Add badges to readme

3m3kalionel committed:
Edit readme badge links

3m3kalionel committed:
Add test file to test script

3m3kalionel committed:
Add response status (201)

3m3kalionel committed:
Add dummy users

3m3kalionel committed:
Edit request

3m3kalionel committed:
Edit response message for adding users

3m3kalionel committed:
Edit authentication error response message

3m3kalionel committed:
Edit error return message

3m3kalionel committed:
Remove error log statement

3m3kalionel committed:
Add new user tests

3m3kalionel committed:
Add dummy test data

3m3kalionel committed:
Add dummy test data

3m3kalionel committed:
Add tests

3m3kalionel committed:
Delete redundant file

3m3kalionel committed:
Delete redundant file

3m3kalionel committed:
Delete redundant file

3m3kalionel committed:
Remove reference to deleted file

3m3kalionel committed:
Delete redundant test

3m3kalionel committed:
Merge pull request #13 from 3m3kalionel/feature/tests

Feature/tests

3m3kalionel committed:
Edit coveralls badge link

3m3kalionel committed:
Merge pull request #15 from 3m3kalionel/feature/tests

Edit coveralls badge link

3m3kalionel committed:
Add code climate badge

3m3kalionel committed:
Add code climate configuration

3m3kalionel committed:
Merge pull request #16 from 3m3kalionel/feature/tests

Feature/tests

3m3kalionel committed:
Edit production database

3m3kalionel committed:
Merge pull request #17 from 3m3kalionel/feature/tests

Edit production database

3m3kalionel committed:
Cover edge cases

3m3kalionel committed:
Edit test response

3m3kalionel committed:
Add secret key

3m3kalionel committed:
Instal jsonwebtoken

3m3kalionel committed:
Set up token for user on sign up

3m3kalionel committed:
Refactor jwt implementation

3m3kalionel committed:
Refactor jwt sign in implementation

3m3kalionel committed:
Use middleware to protect routes

3m3kalionel committed:
Add middleware logic

3m3kalionel committed:
Seperate concerns - remove middleware logic

3m3kalionel committed:
Edit code to fix bug

3m3kalionel committed:
Create user table to fix bug

3m3kalionel committed:
Implement group controller functions using jsonwebtoken

3m3kalionel committed:
Fix primary key bug

3m3kalionel committed:
Fix primary key bug

3m3kalionel committed:
Make group name unique

3m3kalionel committed:
Add group validation middleware

3m3kalionel committed:
Add group validation middleware to routes

3m3kalionel committed:
Delete bug fixing comments

3m3kalionel committed:
Delete bug fixing comments

3m3kalionel committed:
Prevent adding existing user

3m3kalionel committed:
Validate user with middleware

3m3kalionel committed:
Add user validation to routes

3m3kalionel committed:
Add id field

3m3kalionel committed:
Prevent empty messages

3m3kalionel committed:
Add group validation

3m3kalionel committed:
Add group validation

3m3kalionel committed:
Edit commit messages

3m3kalionel committed:
Add middleware for messages api

3m3kalionel committed:
Add route to list group members

3m3kalionel committed:
Add route to list a user's groups

3m3kalionel committed:
Add middleware to validate content

3m3kalionel committed:
Add authenticaion middleware

3m3kalionel committed:
Re-arrange dummy data

3m3kalionel committed:
Merge pull request #18 from 3m3kalionel/feature/authentication

Feature/authentication

3m3kalionel committed:
Edit response message

3m3kalionel committed:
Edit tests

3m3kalionel committed:
Modify files for tests

3m3kalionel committed:
Merge pull request #19 from 3m3kalionel/feature/tests

Feature/tests

3m3kalionel committed:
Add logic input validation

3m3kalionel committed:
Add logic for password input validation

3m3kalionel committed:
Add middleware

3m3kalionel committed:
Merge pull request #20 from 3m3kalionel/feature/authentication

Feature/authentication

3m3kalionel committed:
Add more test cases

3m3kalionel committed:
Merge pull request #21 from 3m3kalionel/feature/tests

Add more test cases

3m3kalionel committed:
Edit readme file

3m3kalionel committed:
Edit response object

3m3kalionel committed:
Edit response object

3m3kalionel committed:
Edit response object

3m3kalionel committed:
Edit response object

3m3kalionel committed:
Edit response message

3m3kalionel committed:
Add middleware to authenticate group

3m3kalionel committed:
Retrieve user id from token

3m3kalionel committed:
Cater for users with no groups

3m3kalionel committed:
Merge pull request #22 from 3m3kalionel/feature/tests

Feature/tests

3m3kalionel committed:
Update README.md

3m3kalionel committed:
Merge branch 'develop' into feature/tests

3m3kalionel committed:
Merge pull request #23 from 3m3kalionel/feature/tests

Update README.md

3m3kalionel committed:
Edit badges

3m3kalionel committed:
Cater for user with no group

3m3kalionel committed:
Fix merge conflicts

3m3kalionel committed:
Merge pull request #24 from 3m3kalionel/feature/tests

Feature/tests

3m3kalionel committed:
Add dialect field to production environment

3m3kalionel committed:
Install packages

3m3kalionel committed:
Undo delete

3m3kalionel committed:
Set up webpack

3m3kalionel committed:
refactor tests

3m3kalionel committed:
Fix test failing issue with webpack

3m3kalionel committed:
Fix random route test

3m3kalionel committed:
Merge pull request #25 from 3m3kalionel/chore/webpack-config

Chore/webpack config

3m3kalionel committed:
chore(dependencies): install client dependencies and devDependencies

3m3kalionel committed:
Set up redux store

3m3kalionel committed:
add action types and creators

3m3kalionel committed:
Set up redux

3m3kalionel committed:
Front end react redux application

3m3kalionel committed:
Merge pull request #27 from 3m3kalionel/feature/react-redux-setup

Feature/react redux setup

3m3kalionel committed:
temp

3m3kalionel committed:
Search for users using a query string

3m3kalionel committed:
Add query logic to cater for empty u

3m3kalionel committed:
Remove redundant code

3m3kalionel committed:
Set up  action and reducers for adding new user

3m3kalionel committed:
Edit logic for  creating mesage success action

3m3kalionel committed:
Add use to group logic

3m3kalionel committed:
Create-group files

3m3kalionel committed:
Add verification to reset password

3m3kalionel committed:
Add field for phone numbers

3m3kalionel committed:
Install package

3m3kalionel committed:
Create client-side route for password reset

3m3kalionel committed:
Implement reset password feature

3m3kalionel committed:
Merge pull request #28 from 3m3kalionel/feature/create-group

Feature/create group

3m3kalionel committed:
Implement the use of JSDoc

3m3kalionel committed:
Merge pull request #29 from 3m3kalionel/fixes/feedback

Implement the use of JSDoc

3m3kalionel committed:
Fix bug in sign up form

3m3kalionel committed:
Create bundle files for production

3m3kalionel committed:
Merge pull request #30 from 3m3kalionel/fixes/feedback

Fixes/feedback

3m3kalionel committed:
chore(feedback): implement defense feedback

- Add reset password feature
- Refactor components
- Improve app styling

[Delivers #150701864]

3m3kalionel committed:
client-side-tests

3m3kalionel committed:
Merge pull request #31 from 3m3kalionel/fixes/feedback

chore(feedback): implement defense feedback

tomily1 commented on .codeclimate.yml:
```diff
@@ -0,0 +1,26 @@
+  paths:
+  - "server/**/*"
+  - "app.js"
+  - "index.js"
```
the red arrow means, there should be a newline at the end of the file

tomily1 commented on client/actions/userActions.js:
```diff
@@ -0,0 +1,107 @@
+import axios from 'axios';
```
why is there space here?

tomily1 commented on client/components/App.jsx:
```diff
@@ -0,0 +1,27 @@
+import DashboardNavbar from './dashboard/DashboardNavbar';
+
+
+class App extends Component {
```
jsDoc here and all other class name

tomily1 commented on client/components/dashboard/ChatArea.jsx:
```diff
@@ -0,0 +1,204 @@
+// import React, { Component } from 'react';
+// import Proptypes from 'prop-types';
```
remove commented or unused code

tomily1 commented on client/components/dashboard/ChatInput.jsx:
```diff
@@ -0,0 +1,70 @@
+  }
+}
+
+// ChatInput.propTypes = {
```
remove commented code

tomily1 commented on client/components/dashboard/MessageList.jsx:
```diff
@@ -0,0 +1,99 @@
+        </li>
+      ));
+    } else {
+      // messageComponent = <div>
```
same here

tomily1 commented on client/store/devConfig.js:
```diff
@@ -0,0 +1,20 @@
+import rootReducer from '../reducers';
+
+const logger = createLogger({
+  // options
```
Are you using this?.... if not, remove

tomily1 commented on template/js/custom.js:
```diff
@@ -0,0 +1,25 @@
+//     $(".dropdown-button").dropdown({ hover: true });
+//     $(".button-collapse").sideNav();
+//     $('.collapsible').collapsible();
+//     $('.modal').modal();
```
remove commented code

3m3kalionel committed:
fix(client-side): implement defence feedback
- create .hound.yml file
- style forgot password password page
- style reset password page
- adjust line character length
- remove redundant code
- remove commented code
- rename components
- remove redundant components
- [Delivers #150799836]

3m3kalionel committed:
- edit .hound.yml configuration

3m3kalionel committed:
fix(build): fix webpack build for production

3m3kalionel committed:
- remove google login/signup button
- [Delivers #150806981]

3m3kalionel committed:
Add Heroku generated app.json

3m3kalionel committed:
feat(client-side): implement google authentication
- install react-google-login and font-awesome packages
- create google authentication api endpoint
- refactor user model
- refator user controller logic
- implement google authentication logic
- implement UI for google authentication
[Finishes #150701865]

3m3kalionel committed:
fix(client-side): implement defence feedback
- add styling for page elements
- add JSDoc
- add tooltips to label components
- include paginiation for searching users
- indicate active group when selected
- add extra api route
- refactor components
[Delivers #150799836]

3m3kalionel committed:
Merge branch 'feature/google-authentication' into fixes/feedback

3m3kalionel committed:
fix(bug fix): implement feedback

- remove comments
- remove console.log statements
- replace void with undefined in JSDoc
- remove redundant validate className
- add priority, time and sender to messages
- add cascade option to drop tables in models
- make component className standard(Dashboard.jsx)
- add styling to elements

[Delivers #150799836]

3m3kalionel committed:
debugging heroku bug

3m3kalionel committed:
Merge pull request #33 from 3m3kalionel/fixes/feedback

fix(client-side): implement defence feedback

3m3kalionel committed:
fix heroku bug

3m3kalionel committed:
Create LICENSE

3m3kalionel committed:
fix(): feedback implementation

- add example.env file
- update readme file
- add in-app notifications
- write PropTypes for validation
- style notifications
- remove redundant src file
- clear modal input state on close

[Delivers #150799836]

3m3kalionel committed:
Merge branch 'fixes/feedback' of https://github.com/3m3kalionel/PostIt into fixes/feedback

3m3kalionel committed:
fix(): fix heroku import bug

- [Delivers #150799836]

3m3kalionel committed:
fix(): fix heroku import bug

- [Delivers #150799836]

3m3kalionel committed:
fix(): bug fix

- style forgot password and reset password pages
- rename GroupsDiv, ChatArea and ChatInput components

[Delivers #151311888]

3m3kalionel committed:
fix(): implement feedback

- style forgot password and reset password forms
- rename components: ChatInput, ChatArea, GroupsDiv
- reduce margin no priority buttons
- hide textarea, priority buttons, addUser buttons when no group is selected
- set up for testing

[Delivers #151311888]

3m3kalionel committed:
fix(): feedback implementation

- remove console.log statements
- rename jsx files using Pascal casing
- style forgot password form
- re-order phone number field in sign up form
- remove commented code
- fix mail verification issue
- move packages to dependences

[Delivers #151521173]

3m3kalionel committed:
chore(): client side tests

- write tests for actions
- write tests for reducers
- write tests for components

[Delivers #151521173]

3m3kalionel committed:
Merge pull request #36 from 3m3kalionel/chore/client-side-tests

Chore/client side tests

3m3kalionel committed:
feat(): implement feedback

- set eslint line character length to 80
- install package
- add action type to clear search list
- remove commented code
- remove console.log statements
- display group members when searching/adding group members
- fix eslint import errors
- fix tooltip issue on logout
- hide submit button when there is no input
- hide text area when no group is selected
- fix eslint JSDoc issues
- refactor code
[Delivers #151521173]

3m3kalionel committed:
Merge branch 'fixes/feedback' into chore/client-side-tests

3m3kalionel committed:
fix(): fix logout tooltip issue

- remove tooltip on logout

[Delivers #151521173]

3m3kalionel committed:
Merge pull request #37 from 3m3kalionel/fixes/feedback

Fixes/feedback
